### DISCUSSION TITLE
Feature request: Support for collection level configurations

### DISCUSSION THREAD

**Discussion by futhwo** [darrenburns/posting#146] [Ideas]
It would be nice to be able to set some options (e.g. authentication, or ssl certificate checking) at collection or folder level instead of at request level, having those options automatically inherited by child objects (requests, other subfolders).

**Comment by darrenburns** [comment]
Totally agree this would be an awesome feature.

I was considering some kind of system where you can create a "base" YAML files in directories, and those would be used to build up the default values when a new request is created.

**Comment by darrenburns** [comment]
I think global config is going to be needed. We would need to think about the UX so that it's not confusing to users. I was thinking a YAML file could be present in each directory and we use inheritance to build up the config for a request.

```
.
|
| foo/
| _base.posting.yaml
| |- bar/
|   |- _base.posting.yaml
|   |- my_request.posting.yaml
```

The config for the `my_request.posting.yaml` would be built by first reading `foo/_base.posting.yaml`, then applying `foo/bar/_base.posting.yaml`, then finally applying the request-specific config. So the config defined in the request file would have highest priority. 

Auth config could be defined in `foo/_base.posting.yaml` and it would apply to the whole collection. If the requests in `foo/bar/` wanted to use a different auth scheme, they could override it in `foo/bar/_base.posting.yaml`.

Something that can help for now is creating requests via the "dupe" and "quick dupe" options - you can create a template request and instead of making new requests via ctrl+n, you can duplicate your template request and the auth/scripts config will be copied over. Not perfect, but faster than doing things manually.

cc @kkHAIKE
